### DISCUSSION TITLE
Pre-generate all image sizes

### DISCUSSION THREAD

**Discussion by johnib** [vercel/next.js#32404] [Ideas]
### Describe the feature you'd like to request

I'm using `next/image` component (and I love it!).
99% of my images are in this format:

```JSX
<Image src={image} alt={imageAlt} placeholder="blur" layout="responsive" />
```

I would like these images, being `imported` and moved to the `next/static` folder to be pre-generated with all the possible **width** values as configured in my `next.config.js` file.

This is to optimize the edge-case where my images were requested with a _width_ value that was never used before, and thus requiring my server generate the images.

### Describe the solution you'd like

You would probably know better than me how it would be best implemented.
If it was a setting that can be enabled in `next.config.js` that would be amazing.

### Describe alternatives you've considered

Running a script to fetch all my images with the various possible **width** values.

**Comment by ghost** [comment]
Agreed, this would be very useful, as generating larger images on the fly can be resource-heavy and slow.

**Reply by johnib** [reply]
When can we expect to get this feature? 😀

**Comment by Tomas2D** [comment]
Has anyone written a custom script for that?
